fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImpl.java

```diff
@@ -266,11 +270,13 @@ public void onThrowable(Throwable t) {
             }
             LOG.error("druid {} http request failed: ", serviceConfig.getNameAndUrl(), t);
             failure.invoke(t);
+            responseFuture = new FailedFuture<>("Failed to reach druid");
         }
+        return responseFuture;
```

Comment:
Rather than capture into a temporary variable and then return, just return the `new FailedFuture` and from `requestBuilder.execute` directly. There's minimal use in the local variable.